src/Symfony/Component/Translation/IdentityTranslator.php

```diff
@@ -11,53 +11,38 @@
 
 namespace Symfony\Component\Translation;
 
+use Symfony\Contracts\Translation\TranslatorTrait;
+
 /**
  * IdentityTranslator does not translate anything.
  *
  * @author Fabien Potencier <fabien@symfony.com>
  */
 class IdentityTranslator implements TranslatorInterface
 {
+    use TranslatorTrait {
+        transChoice as private doTransChoice;
+    }
+
     private $selector;
-    private $locale;
 
     /**
      * @param MessageSelector|null $selector The message selector for pluralization
      */
     public function __construct(MessageSelector $selector = null)
```

Comment:
should we deprecate passing a MessageSelector here ? Or maybe deprecate the class entirely in favor of the IdentityTranslator of the Contracts ?

Reply:
I added a deprecation when a child class calls the constructor of `IdentityTranslator`.